src/MemoryManager.php

```diff
@@ -128,7 +135,7 @@ public function __construct(CacheStorage $cache = null)
         $this->generateMemManagerId();
 
         $memoryLimitStr = trim(ini_get('memory_limit'));
-        if ($memoryLimitStr != '' && $memoryLimitStr != -1) {
+        if ($memoryLimitStr !== '' && $memoryLimitStr !== -1) {
```

Comment:
The reason Ci failing is due to your change of strict comparison. You will need to update `-1` to `'-1'`, like so:
```
if ($memoryLimitStr !== '' && $memoryLimitStr !== -1) {
```
The Continuous integration is setting `memory_limit` to `'-1'`.
```
Run laminas/laminas-continuous-integration-action@v1
[2](https://github.com/laminas/laminas-memory/runs/4809554527?check_suite_focus=true#step:3:2)
  with:
[3](https://github.com/laminas/laminas-memory/runs/4809554527?check_suite_focus=true#step:3:3)
    job: {"command":"./vendor/bin/phpunit","php":"8.1","extensions":[],"ini":["memory_limit        = -1"],"dependencies":"latest","ignore_platform_reqs_8":false,"ignore_php_platform_requirement":false,"additional_composer_arguments":[]}
```
But since this condition was changed to strict comparison, it evaluates to true instead of false, causing the `$this->memoryLimit` to be set to 0 on Line 154.

You can reproduce this on your local machine by setting `memory_limit` in your local `php.ini` to `-1` like so:
```
memory_limit = -1
```

Feel free to cherry-pick my commit here:
https://github.com/laminas/laminas-memory/pull/7/commits/85fd09320689536cfdcff9ed479ac6e1d41bacb8

Reply:
@ATOM594 Many thanks. https://github.com/laminas/laminas-memory/commit/85fd09320689536cfdcff9ed479ac6e1d41bacb8 cherry-picked into my branch.